From 5bc35a9634d20fbd6a6168e589ecdc8b539bee43 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Wed, 24 Jul 2019 22:52:21 +0200
Subject: [PATCH 365/386] telephony: mvebu_phone: update telephony guide to the
 latest sources

Update the documentation with proper defconfig files and
actual logs from the voice-tool, which now requires specifying
the interface under test.

Change-Id: Ieec524a0b034be808f0734e6a0285cb7339782c4
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/13013
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../mvebu_phone/telephony_user_guide.txt      | 169 +-----------------
 1 file changed, 7 insertions(+), 162 deletions(-)

diff --git a/drivers/telephony/mvebu_phone/telephony_user_guide.txt b/drivers/telephony/mvebu_phone/telephony_user_guide.txt
index 8c0da3c8bfcc..81a93df82d25 100644
--- a/drivers/telephony/mvebu_phone/telephony_user_guide.txt
+++ b/drivers/telephony/mvebu_phone/telephony_user_guide.txt
@@ -5,20 +5,19 @@ Introduction
 ------------
 There are 2 telephony controllers in Marvell SoCs:
 	- 2-channel TDM (TDM2C, available in A380)
-	- Multi-channel TDM (TDMMC, available in A8k)
+	- Multi-channel TDM (TDMMC, available in AXP, A7k/8k and CN913x)
 
 In order to use telephony drivers the following SW modules are required
 (Explanation will be provided later is this document):
 
 	- Linux Kernel Image with telephony support
 	- Linux Kernel device-tree with telephony support
-	- SLIC (Subscriber line interface card) Linux Kernel module
 	- Demo user-space application (mv-voice-tool)
 
 Linux Image
 -----------
-arch/arm/configs/mvebu_v7_lsp_defconfig
-arch/arm64/confgs/mvebu_v8_lsp_defconfig
+arch/arm/configs/mvebu_v7_defconfig
+arch/arm64/confgs/marvell_v8_sdk_defconfig
 enable usage of both TDM2C and TDMMC, proper operation mode is fully
 determined by Device Tree entry.
 
@@ -26,30 +25,6 @@ Device-tree
 ------------
 Refer to: Documentation/devicetree/bindings/telephony/mvebu-phone.txt
 
-SLIC
-----
-The LSP release contains pre-compiled Kernel modules for supported SLIC devices,
-located in: rootfs/lib/modules/slic/:
-
-	- Silabs SLICs:
-		- silabs-si3217x.ko
-		- silabs-si3226x.ko
-	- Zarlink SLICs:
-		- zarlink-ve792.ko
-		- zarlink-ve880.ko
-		- zarlink-ve886.ko
-		- zarlink-ve888.ko
-		- zarlink-ve890.ko
-
-The corresponding Kernel module should be inserted after boot using::
-
-	> insmod <Kernel_module_name>
-
-For example::
-
-	> insmod silabs-si3226x.ko
-
-
 Accessing multiple TDM controllers
 ----------------------------------
 In order to make use of the user-space telephony applications, it is
@@ -69,153 +44,23 @@ However, depending on the architecture and requirements, the TAL
   - tal_dev_init/exit()
 allow to register and connect the TDM controllers with other drivers.
 
-mv-voice-tool
--------------
-User-space application which communicates via ioctl with the telephony driver.
-The application is located in the LSP release (rootfs/usr/local/voice-tool).
-mv-voice-tool must suit the SLIC which is used:
-
-	+-----------------------+-----------------------+
-	| SLIC			| mv-voice-tool		|
-	+=======================+=======================+
-	| silabs-si3226x.ko	| mv_voice_tool-si3217x	|
-	+-----------------------+-----------------------+
-	| silabs-si3217x.ko	| mv_voice_tool-si3217x	|
-	+-----------------------+-----------------------+
-	| zarlink-ve792.ko	| mv_voice_tool-ve792	|
-	+-----------------------+-----------------------+
-	| zarlink-ve880.ko	| mv_voice_tool-ve880	|
-	+-----------------------+-----------------------+
-	| zarlink-ve886.ko	| mv_voice_tool-ve886	|
-	+-----------------------+-----------------------+
-	| zarlink-ve888.ko	| mv_voice_tool-ve888	|
-	+-----------------------+-----------------------+
-	| zarlink-ve890.ko	| mv_voice_tool-ve890	|
-	+-----------------------+-----------------------+
-
-
-In order to run mv-voice tool, run the following command::
-
-	> ./mv_voice_tool-<version>
-
-Example for connecting 2 phones using mv-voice-tool::
-
-	> ./mv_voice_tool-si3226x
-
-	>>  Please enter total lines number: 2
-	>>  Please enter PCM sample size(1/2/4): 1
-
-	!!! Remember to start phone devices before performing any action !!!
-
-	Marvell Voice Tool (Silabs Edition):
-	0. Read from SLIC register
-	1. Write to SLIC register
-	2. Start ring
-	3. Stop ring
-	4. Start/Stop HW Dial tone
-	5. Start SW Dial tone
-	6. Self echo on local phone
-	7. Loopback two local phones
-	8. Digital Loopback
-	9. Channel balancing
-	c. Config TDM PCLK
-	m. Multiple local phone pairs loopback
-	t. Start Phone devices
-	u. Stop Phone devices
-	a. Analog Loopback
-	b. record from phone and replay
-	y. Loopback two local phones endlessly for Abacus
-	z. Loopback two local phones for Abacus
-	q. Quit
-
-	>>  Please select option: t
-	[   33.542089] mvebu_phone f47a0000.tdm: Loading Marvell Telephony Driver
-	[   33.570432] ioctl: SILABS_MOD_IOX_CTRL_IF
-	[   33.574470] ioctl: SILABS_MOD_IOX_DEVICE_INIT
-	[   33.578866] ioctl: SILABS_MOD_IOX_CHAN_INIT
-	[   33.583076] ioctl: SILABS_MOD_IOX_CHAN_INIT
-	[   33.587286] ioctl: SILABS_MOD_IOX_CHAN_INIT
-	[   33.591495] ioctl: SILABS_MOD_IOX_CHAN_INIT
-	[   33.595699] ioctl: SILABS_MOD_IOX_CTRL_IF
-	[   33.599734] ioctl: SILABS_MOD_IOX_CHAN_OP
-	[   34.103808] ioctl: SILABS_MOD_IOX_CHAN_OP
-	[   34.607878] ioctl: SILABS_MOD_IOX_CHAN_ALL
-	[   34.612191] Si3226x: Channel 0 : Type = PROSLIC
-	[   34.616917] Si3226x: Channel 1 : Type = PROSLIC
-	[   34.621585] Si3226x: Channel 0 : Chip Type 28
-	[   34.625971] Si3226x: Channel 0 : Chip Rev 3
-	[   34.630289] Si3226x: Channel 1 : Chip Type 28
-	[   34.634672] Si3226x: Channel 1 : Chip Rev 3
-	[   34.639379] Si3226x: loading patch: 08132013
-	[   34.758946] Si3226x: loading patch: 08132013
-	[   35.270230] Si3226x: VBAT @ 50ms = 59.111 v
-	[   35.336159] Si3226x: VBAT Up = 59.111 v
-	[   35.403180] Si3226x: VBAT @ 50ms = 59.635 v
-	[   35.469114] Si3226x: VBAT Up = 59.845 v
-	[   36.141300] ioctl: SILABS_MOD_IOX_CHAN_ALL
-	[   36.145419] Si3226x: Starting LB Cal on channel 0
-	[   38.262691] Si3226x: Starting LB Cal on channel 1
-	[   40.379965] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.393731] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.405775] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.432049] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.439158] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.453907] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.467656] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.479694] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.505975] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.513090] ioctl: SILABS_MOD_IOX_CHAN_SETUP
-	[   40.527837] ioctl: SILABS_MOD_IOX_PCM_TS_SETUP
-	[   40.532526] ioctl: SILABS_MOD_IOX_CHAN_LINE_FEED
-	[   40.537281] ioctl: SILABS_MOD_IOX_PCM_TS_SETUP
-	[   40.541969] ioctl: SILABS_MOD_IOX_CHAN_LINE_FEED
-	[   40.546724] ioctl: SILABS_MOD_IOX_CHAN_OP
-	[   40.551030] ioctl: SILABS_MOD_IOX_CHAN_OP
-	[   40.555294] ioctl: SILABS_MOD_IOX_CHAN_OP
-	[   40.559328] Si3226x: PCMStart
-	[   40.562425] ioctl: SILABS_MOD_IOX_CHAN_OP
-	[   40.566459] Si3226x: PCMStart
-
-	Marvell Voice Tool (Silabs Edition):
-	0. Read from SLIC register
-	1. Write to SLIC register
-	2. Start ring
-	3. Stop ring
-	4. Start/Stop HW Dial tone
-	5. Start SW Dial tone
-	6. Self echo on local phone
-	7. Loopback two local phones
-	8. Digital Loopback
-	9. Channel balancing
-	c. Config TDM PCLK
-	m. Multiple local phone pairs loopback
-	t. Start Phone devices
-	u. Stop Phone devices
-	a. Analog Loopback
-	b. record from phone and replay
-	y. Loopback two local phones endlessly for Abacus
-	z. Loopback two local phones for Abacus
-	q. Quit
-
-	>>  Please select option: 7
-	>>  Enter line #0: 1
-	>>  Enter line #1: 0
-
 generic mv-voice-tool
 ---------------------
 User-space application which communicates via ioctl with the telephony driver.
 The application allows to configure the TDM interface TX/RX lines in
 internal loopback and check data integrity with sending/receiving incremental
-pattern. It doesn't require any external SLIC device and board modifications.
+pattern. It doesn't require any external devices and board modifications.
 
 In order to run mv-voice tool, run the following command::
 
 	# ./mv_voice_tool
 
-Example for connecting test 32 lines with 2-byte PCM sample size::
+Example for testing 32 lines with 2-byte PCM sample size on controller #1::
 
 	# ./mv_voice_tool
 
+	>>  Please enter TDM controller index: 1
+
 	>>  Please enter total lines number: 32
 	>>  Please enter PCM sample size(1/2/4): 2
 
-- 
2.17.1

